src/Core/src/Platform/Android/ButtonExtensions.cs

```diff
@@ -89,75 +64,214 @@ public static void UpdatePadding(this Button platformControl, Thickness padding,
+			}
+			else
+			{
+				var (width, color, radius) = button.GetStrokeProperties(platformView.Context!, true);
+
+				platformView.StrokeColor = color;
+
+				platformView.StrokeWidth = width;
+
+				platformView.CornerRadius = radius;
+			}
+		}
+
+		internal static void UpdateButtonBackground(this MaterialButton platformView, IButton button)
 		{
 			var background = button.Background;
 
-			if (background is GradientPaint)
+			// Get the current background
+			var recreateRipple = false;
+			if (!platformView.TryGetMauiBackground(out var rippleDrawable, out var layerDrawable, out var backgroundDrawable, out var strokeDrawable, out _))
 			{
-				// Remove previous background gradient if any
-				if (platformView.Background is RippleDrawable previousBackground)
+				// The current background is the default one.
+
+				if (background.IsNullOrEmpty())
 				{
-					var previousGradientDrawable = previousBackground.FindDrawableByLayerId(BackgroundDrawableId);
-					previousGradientDrawable?.Dispose();
+					// The new background is empty/default, so no need to do anything.
+					return;
 				}
 
-				// Null out BackgroundTintList to avoid that the MaterialButton custom background doesn't get tinted.
-				platformView.BackgroundTintList = null;
+				// We are going to set a new background, so make a brand new drawable.
+				recreateRipple = true;
+			}
 
-				// Ripple selection color
-				var rippleColor = platformView.RippleColor ?? ColorStateList.ValueOf(Colors.White.WithAlpha(0.5f).ToPlatform());
+			var context = platformView.Context!;
 
-				var w = platformView.Width;
-				var h = platformView.Height;
+			var (width, color, radius) = button.GetStrokeProperties(platformView.Context!, false);
 
-				// Button background gradient
-				GradientDrawable? gradientDrawable = null;
+			// The previous background may have had transparency or a gradient which cannot
+			// be un-set, so we need an entirely new drawable.
+			backgroundDrawable = new GradientDrawable();
+			backgroundDrawable.SetCornerRadius(radius);
+			backgroundDrawable.SetStroke(width, AColor.Transparent);
 
-				if (background is GradientPaint)
-				{
-					gradientDrawable = background.ToGradientDrawable(w, h);
-					gradientDrawable?.SetShape(ShapeType.Rectangle);
-					gradientDrawable?.SetCornerRadius(button.CornerRadius);
-				}
+			if (background is null)
+			{
+				// The new background is null/empty, so we need to revert back to the "default background".
+				// This might be expensive, but only occurs when the background is changed from a non-null
+				// value to null while the app is running.
 
-				if (background is SolidPaint solidPaint)
-				{
-					gradientDrawable = new GradientDrawable();
-					gradientDrawable?.SetShape(ShapeType.Rectangle);
-					gradientDrawable?.SetColor(ColorStateList.ValueOf(solidPaint.Color.ToPlatform()));
-				}
+				// Copy the tints from a temporary button.
+				using var btn = new MaterialButton(context);
```

Comment:
Passing the default drawable has been changed to copy the tints from a temporary button. It is not an operation that happens frequently, set a background and then removing it, but is this perhaps more expensive?

Reply:
It is a bit more expensive, but we don't need to add fields to handlers and pass the default drawable around. That requires new APIs and a bunch of old ones not working. It is more expensive, but this is smei-temporary because all I am using is the tint:

```cs
var defaultTintList = btn.BackgroundTintList;
var defaultColor = defaultTintList?.GetColorForState([R.Attribute.StateEnabled], AColor.Black) ?? AColor.Black;
```

A better option is to not do either and just figure out the default tint color. There is probably some theme resource or some API to just get the raw color. However the color goes from android resources to the main color of the button - some primary color thing. If we can do that in our code, it will just be reading a resource color from the android resources.

I'll create an issue to fix this.

Reply:
https://github.com/dotnet/maui/issues/21579